feat: DNS.MaxCacheTTL for DNS-over-HTTPS resolvers (#8615)
ipfs/go-ipfs: docs/config.md

  ```json
  {
    "eth.": "https://resolver.cloudflare-eth.com/dns-query",
    "crypto.": "https://resolver.cloudflare-eth.com/dns-query"
  }
  ```
  To get all the benefits of a decentralized naming system we strongly suggest setting DoH endpoint to an empty string and running own decentralized resolver as catch-all one on localhost.

Default: `{}`

Type: `object[string -> string]`

### `DNS.MaxCacheTTL`

> Maximum duration entries are valid in the DoH cache.

Reviewer: ```suggestion
Maximum duration that entries are valid in the DoH cache.
```

> This allows to cap the Time-To-Live suggested by the DNS response ([RFC2181](https://datatracker.ietf.org/doc/html/rfc2181#section-8)).

Reviewer: ```suggestion
This allows you to cap the Time-To-Live suggested by the DNS response ([RFC2181](https://datatracker.ietf.org/doc/html/rfc2181#section-8)).
```

If present, the upper bound is applied to DoH resolvers in [`DNS.Resolvers`](#dnsresolvers).

> Note: This does NOT work with OS-level resolver. To make this a global setting, add `.` entry to `DNS.Resolvers` first.

Reviewer: ```suggestion
Note: this does NOT work with Go's default DNS resolver. To make this a global setting, add a `.` entry to `DNS.Resolvers` first.
```

**Examples:**
* `"5m"` DNS entries are kept for 5 minutes or less.
* `"0s"` DNS entries expire as soon as they are retrieved.

Default: Respect DNS Response TTL

> Type: `optionalDuration`

Reviewer: For completeness, can we add this to the `Types` section?

Reviewer 2: Good idea, but opened https://github.com/ipfs/go-ipfs/pull/8729 to not block this PR.
